modcc/kinrewriter.cpp

```diff
@@ -0,0 +1,159 @@
+#include <iostream>
+#include <map>
+#include <string>
+#include <list>
+
+#include "astmanip.hpp"
+#include "symdiff.hpp"
+#include "visitor.hpp"
+
+class KineticRewriter : public BlockRewriterBase {
+public:
+    using BlockRewriterBase::visit;
+
+    KineticRewriter() {}
+    KineticRewriter(scope_ptr enclosing_scope): BlockRewriterBase(enclosing_scope) {}
+
+    virtual void visit(ConserveExpression *e) override;
+    virtual void visit(ReactionExpression *e) override;
+
+protected:
+    virtual void reset() override {
+        BlockRewriterBase::reset();
+        dterms.clear();
+    }
+
+    virtual void finalise() override;
```

Comment:
We don't have an official policy that I am aware of. I would stick with "common" programming spellings like _color_ and -_ize_.